bluesky/suspenders.py

```diff
@@ -97,40 +106,69 @@ def _should_resume(self, value):
         """
         raise NotImplementedError()
 
-    def __call__(self, **kwargs):
-        """
-        Make the class callable so that we can
-        pass it off to the pyepics callback stack.
+    def __call__(self, value, **kwargs):
+        """Make the class callable so that we can
+        pass it off to the ophyd callback stack.
 
-        This expects the massive blob that comes from pyepics
+        This expects the massive blob that comes from ophyd
         """
-        value = kwargs['value']
-
         with self._lock:
-            if self._ev is None:
-                # in the case where either have never been
-                # called or have already fully cycled once
-                if self._should_suspend(value):
-                    self._ev = asyncio.Event(loop=self._loop)
-
-                    self._loop.call_soon_threadsafe(
+            if self._should_suspend(value):
+                self._tripped = True
+                loop = self.RE._loop
+                # this does dirty things with internal state
+                if self._ev is None and self.RE is not None:
+                    self.__make_future()
+                    cb = partial(
                         self.RE.request_suspend,
-                        self._ev.wait())
-            else:
-                if self._should_resume(value):
-                    ev = self._ev.set
-                    sleep = self._sleep
+                        self._ev.wait(),
+                        pre_plan=self._pre_plan,
+                        post_plan=self._post_plan)
+
+                    loop.call_soon_threadsafe(cb)
+            elif self._should_resume(value):
+                self._tripped = False
+                self.__set_event()
+
+    def __make_future(self):
```

Comment:
because this is something that subclasses should not be messing with.  I used it in the mds/fs objects as well to hide the 'direct' collection objects. 

This code is a bit fiddly and the class in intended to be sub-classed, so hide the guts.